docs/starting/tooling.md

```diff
@@ -1,78 +1,172 @@
+
+#### Using Docker Compose
+
+Projects which are delivered under the OpenEuropa Initiative offers a set of containers to mimic software packages used on the IT infrastructure and provide an easy way of setting up an environment to work. If you want to run the containerised environment in order to work with a given component, project or site, you can follow these steps to set up it using Docker Compose.
+
+Requirements:
+
+- [Docker](https://www.docker.com/get-docker)
+- [Docker-compose](https://docs.docker.com/compose)
+
+Run:
+
+```bash
+docker-compose up -d
+```
+
+This will set up and run the environment. After spawning, please follow the set of commands specified in the documentation of a given component, site or a project.
+Usually the next step is to execute Composer script to download and set up dependencies.
+
+```bash
+docker-compose exec web composer install
+```
+
+In order to tweak settings or adjust configuration of a specific container, please edit the `docker-compose.yml` file accordingly to your current needs.
+
+#### Using Docker on macOS
+
+In order to maximise the performance of using Docker on macOS, we strongly advise using at least version 18.03.1 of Docker which supports native NFS integration. You can find a very comprehensive article about this topic under following link [Set Up Docker For Mac with Native NFS](https://medium.com/@sean.handley/how-to-set-up-docker-for-mac-with-native-nfs-145151458adc).
+
+Basically, what is required can be defined in two steps.
+
+First is to run the bash script in order to configure Docker native NFS support.
+You can find the source of a script in the article or by using this link [setup_native_nfs_docker_osx.sh](https://gist.githubusercontent.com/seanhandley/7dad300420e5f8f02e7243b7651c6657/raw/fdd77fe66cf9ce893fa0175d735cbede2bb065e4/setup_native_nfs_docker_osx.sh).
+
+Next, you must include an additional volume definition inside the `docker-compose.yml` file.
+
+You can find the example of how to do so below. In the additional part `volumes`, the volume `nfsmount` is defined and configured to use NFS native support. Then you can reference the defined volume as a storage of a container.
+
+In the example below the `nfsmount` volume is used as a storage for the `web` container and mount to the `var/www/html` directory inside the container. From the local machine perspective it mounts the main directory where the `docker-compose.yml` file is located into the directory configured in a give container (`/var/www/html` in this particular case).
+
+```yml
+# Example of the docker-compose.yml file with the additional volume definition.
+
+version: '2'
+services:
+  web:
+    image: fpfis/httpd-php-dev:5.6
+    working_dir: /var/www/html
+    environment:
+      - DOCUMENT_ROOT=/var/www/html
+    volumes:
+      - nfsmount:/var/www/html
+    ports:
+      - 8080:8080
+
+volumes:
```

Comment:
We could maybe add this as commented on all our `docker-compose.yml` files linking to this document for more information. We could rename `nfsmount` into a more generic name, like `project_root` and then configure the volume here so that it can be used by default in Linux and then, by uncommenting that section, on macosx.

Reply:
I would like to see us have a docker-compose.yml.dist in git and a docker-compose.yml which contains local configuration, in the same way we do with runner.yml and behat.yml. This would prevent the need to edit docker-compose.yml on every build. The [task runner|https://github.com/openeuropa/task-runner] and each component would need ammending to suit.

Reply:
Maybe we can use [this line](https://github.com/openeuropa/task-runner/pull/65/files#diff-593ebe9d8282520d2fa084287134bd44R137) to provide a user configuration file for Task Runner ?

Reply:
I know that it would be great @richardcanoe but the problem is that first, you have to spawn the environment in order to use the task runner. We could investigate another tool which could give us the possibility to execute tasks before spawning containers but I guess it's out of the scope for this ticket. Regarding the comment of aligning all components and commenting a part of the volume definition (mentioned in the PR), I agree that it's a good idea but again I think that it should be covered under follow-up ticket that I could prepare.